server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
```

Comment:
can u point me to that coordinator code that sends 404 on not finding datasource. I see a bunch of (in DatasourcesResource.java)
if(dataSource == null) Response.noContent().build()
that would be "204" and not "404" or am i looking in the wrong place?